source/common/router/config_impl.cc

```diff
@@ -69,29 +69,71 @@ ShadowPolicyImpl::ShadowPolicyImpl(const envoy::api::v2::RouteAction& config) {
   runtime_key_ = config.request_mirror_policy().runtime_key();
 }
 
+class HeaderHashImpl : public HashPolicyImpl::HashImpl {
+public:
+  HeaderHashImpl(const std::string& header_name) : header_name_(header_name) {}
+
+  ~HeaderHashImpl() override {}
+
+  Optional<uint64_t> evaluate(const std::string&, const Http::HeaderMap& headers) const override {
+    Optional<uint64_t> hash;
+
+    const Http::HeaderEntry* header = headers.get(header_name_);
+    if (header) {
+      hash.value(HashUtil::xxHash64(header->value().c_str()));
+    }
+    return hash;
```

Comment:
Is it worth the complexity to have this optional? Why not just return a zero hash value if missing?

Reply:
My intent was to remain compatible with the existing interface. Zero is a valid hash value, even if it doesn't show up often.